boards/cats_rev1Pro/Core/Src/tasks/task_state_est.c

```diff
@@ -234,88 +247,95 @@ inline static float calculate_height(float pressure_initial, float pressure, flo
+      }
+      for (uint8_t i = NUM_IMU; i < (NUM_IMU + NUM_ACCEL); i++) {
+        state_data->acceleration[i] = (float)(global_accel.acc_y) / (1024) * GRAVITY / calibration->angle - GRAVITY;
+      }
       break;
     case 2:
       /* Choose Z Axis */
-      state_data->acceleration[0] = (float)(global_imu[0].acc_z) / (1024) * GRAVITY / calibration->angle - GRAVITY;
-      state_data->acceleration[1] = (float)(global_imu[1].acc_z) / (1024) * GRAVITY / calibration->angle - GRAVITY;
-      state_data->acceleration[2] = (float)(global_imu[2].acc_z) / (1024) * GRAVITY / calibration->angle - GRAVITY;
+      /* Fill up state data with IMU and once this is done, continue filling up with accel data */
+      for (uint8_t i = 0; i < NUM_IMU; i++) {
+        state_data->acceleration[i] = (float)(global_imu[i].acc_z) / (1024) * GRAVITY / calibration->angle - GRAVITY;
+      }
+      for (uint8_t i = NUM_IMU; i < (NUM_IMU + NUM_ACCEL); i++) {
+        state_data->acceleration[i] = (float)(global_accel.acc_z) / (1024) * GRAVITY / calibration->angle - GRAVITY;
+      }
       break;
     default:
       break;
   }
 
-  state_data->pressure[0] = (float)(global_baro[0].pressure);
-  state_data->pressure[1] = (float)(global_baro[1].pressure);
-  state_data->pressure[2] = (float)(global_baro[2].pressure);
-
-  state_data->temperature[0] = (float)global_baro[0].temperature / 100.f;
-  state_data->temperature[1] = (float)global_baro[1].temperature / 100.f;
-  state_data->temperature[2] = (float)global_baro[2].temperature / 100.f;
+  for (uint8_t i = 0; i < NUM_BARO; i++) {
+    state_data->pressure[i] = (float)(global_baro[i].pressure);
+    state_data->temperature[i] = (float)global_baro[i].temperature / 100.f;
+  }
 
   /* Add Sensor Noise if asked to */
-  if (fsm_state->flight_state == THRUSTING_1) {
+  // if (*fsm_state == THRUSTING_1) {
 #ifdef INCLUDE_NOISE
-    float rand_pressure[3] = {0};
-    float rand_acc[3] = {0};
-    rand_pressure[0] = PRESSURE_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-    rand_pressure[1] = PRESSURE_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-    rand_pressure[2] = PRESSURE_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-    rand_acc[0] = ACC_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-    rand_acc[1] = ACC_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-    rand_acc[2] = ACC_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
-
-    state_data->pressure[0] += rand_pressure[0];
-    state_data->pressure[1] += rand_pressure[1];
-    state_data->pressure[2] += rand_pressure[2];
-    state_data->acceleration[0] += rand_acc[0];
-    state_data->acceleration[1] += rand_acc[1];
-    state_data->acceleration[2] += rand_acc[2];
+  float rand_pressure[3] = {0};
+  float rand_acc[3] = {0};
+  rand_pressure[0] = PRESSURE_NOISE_MAX_AMPL * ((float)rand() - 2147483648 / 2) / (2147483648 / 2);
```

Comment:
How is rand() seeded? Can it be that every time you reset the board you will always get the same sequence of random numbers?